test/Rentals.spec.ts

```diff
@@ -1966,6 +2134,87 @@ describe('Rentals', () => {
+
+      const bytes = ethers.utils.defaultAbiCoder.encode([offerEncodeType], [offerEncodeValue])
+
+      await land.connect(lessor)['safeTransferFrom(address,address,uint256,bytes)'](lessor.address, rentals.address, tokenId, bytes)
+
+      listingParams = { ...listingParams, maxDays: [rentalDays2], minDays: [rentalDays2], nonces: [0, 0, 1] }
+      acceptListingParams = { ...acceptListingParams, rentalDays: rentalDays2, index: 0 }
+
+      await rentals
+        .connect(tenant)
+        .acceptListing(
+          { ...listingParams, signature: await getListingSignature(lessor, rentals, listingParams) },
+          acceptListingParams.operator,
+          acceptListingParams.index,
+          acceptListingParams.rentalDays,
+          acceptListingParams.fingerprint
+        )
+
+      await network.provider.send('evm_setAutomine', [true])
+
+      await evmMine()
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(lessor.address)
+      expect(rental.tenant).to.equal(tenant.address)
+      expect(rental.endDate).to.equal(latestBlockTimestamp + daysToSeconds(rentalDays1 + rentalDays2))
+    })
+
+    it('should allow calling accept offer after sending the asset on the same block if it is an extension', async () => {
+      const rentalDays1 = 10
+      const rentalDays2 = 100
+
+      let rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(zeroAddress)
+      expect(rental.tenant).to.equal(zeroAddress)
+      expect(rental.endDate).to.equal(0)
+
+      await network.provider.send('evm_setAutomine', [false])
+
+      offerEncodeValue[rentalDaysIndex] = rentalDays1
+      offerEncodeValue[signatureIndex] = await getOfferSignature(tenant, rentals, { ...offerParams, rentalDays: offerEncodeValue[rentalDaysIndex] })
+
+      const bytes = ethers.utils.defaultAbiCoder.encode([offerEncodeType], [offerEncodeValue])
+
+      await land.connect(lessor)['safeTransferFrom(address,address,uint256,bytes)'](lessor.address, rentals.address, tokenId, bytes)
+
+      offerParams = { ...offerParams, rentalDays: rentalDays2, nonces: [0, 0, 1] }
+
+      await rentals.connect(lessor).acceptOffer({ ...offerParams, signature: await getOfferSignature(tenant, rentals, offerParams) })
+
+      await network.provider.send('evm_setAutomine', [true])
+
+      await evmMine()
+
+      const latestBlockTimestamp = await getLatestBlockTimestamp()
+
+      rental = await rentals.rentals(listingParams.contractAddress, listingParams.tokenId)
+
+      expect(rental.lessor).to.equal(lessor.address)
+      expect(rental.tenant).to.equal(tenant.address)
+      expect(rental.endDate).to.equal(latestBlockTimestamp + daysToSeconds(rentalDays1 + rentalDays2))
```

Comment:
Can we add also tests when the block is not minted but the sender is trying to reuse the signature?

Reply:
https://github.com/decentraland/rentals-contract/pull/23/files#diff-3369bd7d295f9e1649f509622263f562231b444e193ccbdd316915f134efd28fR1404 That is verified here

Reply:
There are various tests like this for accept offer, accept listing and accepting offer by sending estate and land in different combinations